File: LeetCode_Algorithm/997(Find the Town Judge)/997.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        # 998ms 38% 18.9MB 90%
        arr = [0] * (n+1)
        for [i, j] in trust:
            arr[i] -= 1
            arr[j] += 1
        for i in range(1, n+1):
            if arr[i] == n-1:
                return i
        return -1

File: LeetCode_Algorithm/997(Find the Town Judge)/997.js
var findJudge = function(n, trust) {
  // sol1 101ms 84% 50.6MB 67%
  let arr = new Array(n+1).fill(0)
  for (let [i,j] of trust) {
      arr[i] -= 1
      arr[j] += 1
  }
  for (let i=1; i <= n; i++) {
      if (arr[i] == n-1)
          return i
  }
  return -1
};